src/systemctl/systemctl.c

```diff
@@ -4168,14 +4197,37 @@ static int status_property(const char *name, sd_bus_message *m, UnitStatusInfo *
                         if (r < 0)
                                 return bus_log_parse_error(r);
 
+                        c = new0(UnitCondition, 1);
+                        if (!c)
+                                return log_oom();
+
                         while ((r = sd_bus_message_read(m, "(sbbsi)", &cond, &trigger, &negate, &param, &state)) > 0) {
                                 log_debug("%s %d %d %s %d", cond, trigger, negate, param, state);
-                                if (state < 0 && (!trigger || !i->failed_condition)) {
-                                        i->failed_condition = cond;
-                                        i->failed_condition_trigger = trigger;
-                                        i->failed_condition_negate = negate;
-                                        i->failed_condition_parameter = param;
-                                }
+
+                                c->name = strdup(cond);
+                                if (!c->name)
+                                        goto on_error;
+
+                                c->trigger = trigger;
+                                c->negate = negate;
+                                c->param = strdup(param);
+                                if (!c->param)
+                                        goto on_error;
+                                c->tristate = state;
+
+                                LIST_PREPEND(condition, i->condition, c);
+
+                                c = new0(UnitCondition, 1);
+                                if (c)
+                                        continue;
+
+                        on_error:
+                                if (c->name)
+                                        free(c->name);
```

Comment:
I think, above new() was failed, on_error is processed. c->name and c->param are dereferenced and return with log_oom() and the c is also dereferenced by _cleanup_free_.
Am I wrong?